src/Servers/IIS/AspNetCoreModuleV2/CommonLib/ServerErrorHandler.h

```diff
@@ -9,107 +9,53 @@
             return;
         }
 
         HTTP_DATA_CHUNK dataChunk = {};
-        IHttpResponse* pResponse = pContext.GetResponse();
-        pResponse->SetStatus(m_statusCode, m_statusText.c_str(), m_subStatusCode, hr, nullptr, true);
+        IHttpResponse* pResponse = m_pHttpContext.GetResponse();
+        pResponse->SetStatus(m_statusCode, m_statusText.c_str(), m_subStatusCode, m_HR, nullptr, true);
         pResponse->SetHeader("Content-Type",
             "text/html",
             (USHORT)strlen("text/html"),
             FALSE
         );
 
         dataChunk.DataChunkType = HttpDataChunkFromMemory;
-        if (m_ExceptionInfoContent.size() > 0)
-        {
-            dataChunk.FromMemory.pBuffer = &m_ExceptionInfoContent[0];
-            dataChunk.FromMemory.BufferLength = static_cast<ULONG>(m_ExceptionInfoContent.size());
-        }
-        else
-        {
-            static std::string s_html500Page = GetHtml(m_moduleInstance, m_page);
-            dataChunk.FromMemory.pBuffer = s_html500Page.data();
-            dataChunk.FromMemory.BufferLength = static_cast<ULONG>(s_html500Page.size());
-        }
-      
-        pResponse->WriteEntityChunkByReference(&dataChunk);
-    }
+        dataChunk.FromMemory.pBuffer = m_ExceptionInfoContent.data();
+        dataChunk.FromMemory.BufferLength = static_cast<ULONG>(m_ExceptionInfoContent.size());
 
-    std::string
-    GetHtml(HMODULE module, int page)
-    {
-        try
-        {
-            HRSRC rc = nullptr;
-            HGLOBAL rcData = nullptr;
-            std::string data;
-            const char* pTempData = nullptr;
-
-            THROW_LAST_ERROR_IF_NULL(rc = FindResource(module, MAKEINTRESOURCE(page), RT_HTML));
-            THROW_LAST_ERROR_IF_NULL(rcData = LoadResource(module, rc));
-            auto const size = SizeofResource(module, rc);
-            THROW_LAST_ERROR_IF(size == 0);
-            THROW_LAST_ERROR_IF_NULL(pTempData = static_cast<const char*>(LockResource(rcData)));
-            data = std::string(pTempData, size);
-
-            auto additionalErrorLink = Environment::GetEnvironmentVariableValue(L"ANCM_ADDITIONAL_ERROR_PAGE_LINK");
-            std::string additionalHtml;
-
-            if (additionalErrorLink.has_value())
-            {
-                additionalHtml = format("<a href=\"%S\"> <cite> %S </cite></a> and ", additionalErrorLink->c_str(), additionalErrorLink->c_str());
-            }
-
-            return format(data, additionalHtml.c_str());
-        }
-        catch (...)
-        {
-            OBSERVE_CAUGHT_EXCEPTION();
-            return "";
-        }
+        pResponse->WriteEntityChunkByReference(&dataChunk);
     }
 
-    IHttpContext& m_pContext;
     HRESULT m_HR;
     bool m_disableStartupPage;
-    int m_page;
-    HINSTANCE m_moduleInstance;
     USHORT m_statusCode;
     USHORT m_subStatusCode;
     std::string m_statusText;
-    std::vector<byte> m_ExceptionInfoContent;
+    std::string& m_ExceptionInfoContent;
```

Comment:
I'd just like to publicly say "I'm not a fan of this!"

Reply:
Allocationsssss